opentelemetry-instrumentation/src/opentelemetry/instrumentation/instrumentor.py

```diff
@@ -65,6 +84,9 @@ def instrument(self, **kwargs):
         ``opentelemetry-instrument`` command does.
         """
 
+        if self._validate_dependencies():
+            return
+
```

Comment:
Do not instrument if the instrumentation is incompatible with the installed version of the target library or if the target library is not installed.